### PR TITLE
Assert 250 Bits and SPLIT_FELT

### LINKED ISSUE #194: `ASSERT_250_BITS` and `SPLIT_FELT` hints

### PR DESCRIPTION
**Pull request by jrchatruc**
Closes #194

### CODE REVIEW

File: pkg/hints/math_hints.go [hunk 1]
```diff
@@ -151,3 +151,118 @@ func sqrt(ids IdsManager, vm *VirtualMachine) error {
...
+	shift, ok := (*constants)[SHIFT]
+
+	if !ok {
+		var err error
+		shift, err = ids.GetConst("SHIFT", constants)
+
+		if err != nil {
+			return err
+		}
+	}
```
Comment (fmoletta) [inline pkg/hints/math_hints.go]: The `GetConst` method searches the accessible scopes from innermost to outermost, yielding correct results as long as the compiled json is valid, so we can skip checking for the common library path first

File: pkg/hints/math_hints.go [hunk 2]
```diff
@@ -151,3 +151,118 @@ func sqrt(ids IdsManager, vm *VirtualMachine) error {
...
+//	assert value < ids.UPPER_BOUND, f'{value} is outside of the range [0, 2**250).'
+//	# Calculation for the assertion.
+//	ids.high, ids.low = divmod(ids.value, ids.SHIFT)
+func Assert250Bit(ids IdsManager, vm *VirtualMachine, constants *map[string]Felt) error {
+	const UPPER_BOUND = "starkware.cairo.common.math.assert_250_bit.UPPER_BOUND"
+	const SHIFT = "starkware.cairo.common.math.assert_250_bit.SHIFT"
+
+	if constants == nil {
+		return errors.New("Called Assert250Bit with a nil constants map")
+	}
```
Comment (fmoletta) [inline pkg/hints/math_hints.go]: I think this check could be redundant, this method is only called during `ExecuteHint` which is only called during `Step` and so on, the we should be able to trust that the constants reference is not nil  as we also trust that the vm reference is not nil

File: pkg/hints/math_hints.go (hunk 1 above)
Comment (jrchatruc) [reply pkg/hints/math_hints.go]: Nice, removed the common library checks here [57cffcd](https://github.com/lambdaclass/cairo-vm_in_go/pull/233/commits/57cffcdae746aac0f48ace7b5b28b1c8343d6b18)

File: pkg/hints/math_hints.go (hunk 2 above)
Comment (jrchatruc) [reply pkg/hints/math_hints.go]: Makes sense, removed here [476ab88](https://github.com/lambdaclass/cairo-vm_in_go/pull/233/commits/476ab88dbab2305ceedd3209486f06d14270616a)

**Review by fmoletta — APPROVED** [review]

**Review by Juan-M-V — APPROVED** [review]

**Pull request merged**